plugins/inputs/marklogic/marklogic.go

```diff
@@ -0,0 +1,233 @@
+				CoreThreads struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"core-threads"`
+				TotalCPUStatUser   float64 `json:"total-cpu-stat-user"`
+				TotalCPUStatNice   float64 `json:"total-cpu-stat-nice"`
+				TotalCPUStatSystem float64 `json:"total-cpu-stat-system"`
+				TotalCPUStatIdle   float64 `json:"total-cpu-stat-idle"`
+				TotalCPUStatGuest  float64 `json:"total-cpu-stat-guest"`
+				MemoryProcessSize  struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-process-size"`
+				MemoryProcessRss struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-process-rss"`
+				MemorySystemTotal struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-system-total"`
+				MemorySystemFree struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-system-free"`
+				DataDirSpace struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"data-dir-space"`
+				QueryReadBytes struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"query-read-bytes"`
+				QueryReadLoad struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"query-read-load"`
+				HTTPServerReceiveBytes struct {
+					Units string  `json:"units"`
+					Value float64 `json:"value"`
+				} `json:"http-server-receive-bytes"`
+				HTTPServerSendBytes struct {
+					Units string  `json:"units"`
+					Value float64 `json:"value"`
+				} `json:"http-server-send-bytes"`
+			} `json:"status-detail"`
+		} `json:"status-properties"`
+	} `json:"host-status"`
+}
+
+func (c *Marklogic) Description() string {
+	return "Gathers host health status data from a Marklogic Cluster"
+}
+
+var sampleConfig = `
+  ## List URLs of Marklogic hosts using Management API endpoint.
+  # hosts = ["http://localhost:8002/manage/v2/hosts/${hostname}?view=status&format=json"]
+
+	# Using HTTP Digest Authentication.
+	# digest_username = "telegraf"
+	# digest_password = "p@ssw0rd"
+`
+
+func (c *Marklogic) SampleConfig() string {
+	return sampleConfig
+}
+
+// Gather read stats from all hosts configured in cluster.
+func (c *Marklogic) Gather(accumulator telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	if len(c.Hosts) == 0 {
+		c.Hosts = []string{"http://localhost:8002/manage/v2/hosts/ml1.local?view=status&format=json"}
+	}
+
+	// Range over all ML servers, gathering stats. Returns early in case of any error.
+	for _, u := range c.Hosts {
+		wg.Add(1)
+		go func(host string) {
+			defer wg.Done()
+			if err := c.fetchAndInsertData(accumulator, host); err != nil {
+				accumulator.AddError(fmt.Errorf("[host=%s]: %s", host, err))
+			}
+		}(u)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+func (c *Marklogic) fetchAndInsertData(acc telegraf.Accumulator, host string) error {
+	if c.client == nil {
+		c.client = &http.Client{
```

Comment:
Doesn't look like we are using the client?  Should we use the `dac.NewTransport` as the client Transport?  Then you can use the http.Client as normal instead of calling RoundTrip.

Reply:
@danielnelson - Decided to switch and  add basic auth over digest so this should be cleaner now - #5590